### DISCUSSION TITLE
Using RAUC to update external Embedded system

### DISCUSSION THREAD

**Discussion by Bouchellal** [rauc/rauc#1467] [Q&A]
Hey, i have the following setup: 

Rasberry-like running on linux and an MCU connected to it via programming link.

I implemented a rauc bundle to update the slot A & B of my Rasberry-lik system. It works fine.
I was looking for a way to add the MCU's firmware to the rauc bundle, to be able to update the MCU via rauc ??

all i need to do is to add the firmware and the flash_mc_script.sh to the bundle and tell rauc to run the flash_mcu_script.sh. Or maybe add the flash_mcu_script to the hook.sh.

Is it doable with RAUC or is it out of RAUC's perimeter ?

**Comment by ejoerns** [comment]
@Bouchellal Maybe have a look at https://rauc.readthedocs.io/en/latest/advanced.html#updating-sub-devices

**Reply by Bouchellal** [reply]
Thank you so much, 

I see that it is not recommended but doable. Where can i find guidelines to implement it (manifest, script, rauc cmd)?

**Reply by jluebbe** [reply]
You can use [slot install hooks](https://rauc.readthedocs.io/en/latest/using.html#slot-hooks) to define how a slot is installed by having a script in the bundle.

**Reply by Bouchellal** [reply]
I have done this :

- Added a new slot [image.mcu_firmware]
- Linked filename=my_firmware.bin
- Defined its hooks=install
- Added slot-install case script in the hook.sh
- Generated successfully the bundle, i can see my_firmware.bin and its sha256sum and size

When it came time to "rauc install <bundle>", i have this error:
`LastError: Installation error: No target slot for class mcu_firmware of image my_firmware.bin`

here is full manifest : 
``` ini
[update]
compatible=wwap
version=2.0
description=Rauc update Bundle
build=20240718093120

[bundle]
format=verity

[hooks]
filename=hook.sh

[image.rootfs]
sha256=578bf50bcd68e0009093834ba2c9c027a81da1525848429c61fa0ce49649e4a3
size=2164400128
filename=production-image.ext4.verity
hooks=pre-install;post-install;

[image.mcu_firmware]
sha256=4efe71834b0058197954057dab89d778409ce424e7bfe8517dbd685cc3aed674
size=16038145
filename=my_firmware.bin
hooks=install;
```

It appears to me that I'am doing sometging wrong, can you help me please?

**Reply by jluebbe** [reply]
You still need to define a matching slot in your system.conf. Perhaps just pointing at /dev/null.

Alternatively, you could not define an image for `my_firmware.bin` in the manifest and instead install if from the `post-install` hook of your rootfs.
